dbms/src/Storages/DeltaMerge/tests/DMTestEnv.h

```diff
@@ -285,13 +299,7 @@ class DMTestEnv
             for (size_t i = 0; i < num_rows; i++)
             {
                 Int64 value = reversed ? end - 1 - i : beg + i;
-                WriteBufferFromOwnString ss;
-                for (size_t index = 0; index < rowkey_column_size; index++)
-                {
-                    ::DB::EncodeUInt(static_cast<UInt8>(TiDB::CodecFlagInt), ss);
-                    ::DB::EncodeInt64(value, ss);
-                }
-                values.emplace_back(ss.releaseStr());
+                values.emplace_back(genMockCommonHandle(value, rowkey_column_size));
```

Comment:
there is one question makes me confused:
in this case, any difference among `values.emplace_back(genMockCommonHandle(value, rowkey_column_size));`,  `values.emplace_back(std::move(genMockCommonHandle(value, rowkey_column_size)));` and `values.push_back(genMockCommonHandle(value, rowkey_column_size));`? Which one is the most effective?

Reply:
There are two questions:
1. `push_back` vs `emplace_back`.
2. `genMockCommonHandle(value, rowkey_column_size)` vs `std::move(genMockCommonHandle(value, rowkey_column_size))`.

The first question, you can refer the cpp reference: [push_back](https://en.cppreference.com/w/cpp/container/vector/push_back) and [emplace_back](https://en.cppreference.com/w/cpp/container/vector/emplace_back). In short, `emplace_back` use placement-new to construct a object of the container. Strictly, `emplace_back` can be more effective than `push_back`.

The answer of the second question is it depends on the return value of the function. In the case of `genMockCommonHandle`, there is no different. 

Because `genMockCommonHandle` returns a unnamed temporary variable(`std::string`) and it is a rvalue and it is moveable (Don't need a `std::move`).

Reply:
thank you, the answer make it clear~